packages/pages/src/util/runtime.test.ts

```diff
@@ -46,104 +57,53 @@ describe("runtime", () => {
     const runtime = getRuntime();
 
     expect(runtime.isServerSide).toBe(true);
+
     global.process = originalProcess;
   });
 
   it("correctly identifies deno via Deno object", async () => {
     const originalProcess = process;
     (global.process as any) = undefined;
-
-    const originalWindow = { ...window };
-    const windowSpy = jest.spyOn(global, "window", "get");
-
-    windowSpy.mockImplementation(
-      () =>
-        ({
-          ...originalWindow,
-          Deno: {
-            version: {
-              deno: "1.24.0",
-            },
-          },
-        } as any)
-    );
+    global.Deno = {
```

Comment:
If we used the third party lib, I think you could mock the third party lib itself with jest.mock() instead of having to manually specify these globals? The benefit there is that we don't have to manually mock low level details like the contents of global.Deno, which could change or be slightly different than we expect. The downside is that we have to trust *somebody else* to do that, but the lib Nidhi found has pretty good usage so I think it's trustable?

Reply:
If you can imagine a scenario where a new version of Deno changes what stuff they provide in the global context, it would be annoying to have to come back and have to retest and update things.

Reply:
I believe mocking the module will be a mess. Once mocked to return true for `isNode` or `isDeno`, we'd still have to mock the global variable because they are read inside the if statement to get the version i.e. `if (isDeno) this.version = Deno.version.deno;`. I would rather just mock the global.
While their globals changing would break the tests, it would be a major breaking change (node 20 or deno 2) and we wouldn't support those right away anyways

Reply:
I believe you can just call jest.mock() at the top and then jest.spyOn().mockImplementation() within each test case?
I don't think it should be messy since we regularly mock search-headless-react's useAnswersState in the component library.
https://github.com/yext/search-ui-react/blob/main/tests/components/DirectAnswer.test.tsx